zipline/data/hdf5_daily_bars.py

```diff
@@ -0,0 +1,634 @@
+from functools import partial
+
+import h5py
+import logbook
+import numpy as np
+import pandas as pd
+from six.moves import reduce
+
+from zipline.data.bar_reader import NoDataBeforeDate, NoDataAfterDate
+from zipline.data.session_bars import SessionBarReader
+from zipline.utils.memoize import lazyval
+
+
+log = logbook.Logger('HDF5DailyBars')
+
+
+DATA = 'data'
+INDEX = 'index'
+LIFETIMES = 'lifetimes'
+
+SCALING_FACTOR = 'scaling_factor'
+
+OPEN = 'open'
+HIGH = 'high'
+LOW = 'low'
+CLOSE = 'close'
+VOLUME = 'volume'
+
+FIELDS = (OPEN, HIGH, LOW, CLOSE, VOLUME)
+
+DAY = 'day'
+SID = 'sid'
+
+START_DATE = 'start_date'
+END_DATE = 'end_date'
+
+
+scaling_factors = {
+    # Retain 3 decimal places for prices.
+    OPEN: 1000,
+    HIGH: 1000,
+    LOW: 1000,
+    CLOSE: 1000,
+    # Volume is expected to be a whole integer.
+    VOLUME: 1,
+}
+
+
+def coerce_to_uint32(a, field):
+    """
+    Returns a copy of the array as uint32, applying a scaling factor to
+    maintain precision if necessary.
+    """
+    return (a * scaling_factors[field]).astype('uint32')
+
+
+def days_and_sids_for_frames(frames):
+    """
+    Returns the date index and sid columns shared by a list of dataframes,
+    ensuring they all match.
+
+    Parameters
+    ----------
+    frames : list[pd.DataFrame]
+        A list of dataframes indexed by day, with a column per sid.
+
+    Returns
+    -------
+    days : np.array[datetime64[ns]]
+        The days in these dataframes.
+    sids : np.array[int64]
+        The sids in these dataframes.
+
+    Raises
+    ------
+    ValueError
+        If the dataframes passed are not all indexed by the same days
+        and sids.
+    """
+    frame_0 = frames[0]
+
+    # Ensure the indices and columns all match.
+    for frame_1 in frames[1:]:
+        if not frame_0.index.equals(frame_1.index):
+            raise ValueError('Frames have mistmatched days.')
+
+        if not frame_0.columns.equals(frame_1.columns):
+            raise ValueError('Frames have mismatched sids.')
+
+    return frame_0.index.values, frame_0.columns.values
+
+
+class HDF5DailyBarWriter(object):
+    """
+    Class capable of writing daily OHLCV data to disk in a format that
+    can be read efficiently by HDF5DailyBarReader.
+
+    Parameters
+    ----------
+    filename : str
+        The location at which we should write our output.
+    date_chunk_size : int
+        The number of days per chunk in the HDF5 file. If this is
+        greater than the number of days in the data, the chunksize will
+        match the actual number of days.
+
+    See Also
+    --------
+    zipline.data.hdf5_daily_bars.HDF5DailyBarReader
+    """
+    def __init__(self, filename, date_chunk_size, driver=None):
```

Comment:
Ah, I was playing around with this in case we wanted to use a different driver to make testing easier. Going to remove this for now, should be easy enough to add back if we do want it.